### ISSUE TITLE
Incorrect additionalTextEdit applied when auto-importing PHP namespaces

### ISSUE THREAD

**Issue by pyr0hu** [sublimelsp/LSP#757]
* OS and language server
  * macOS 10.15 - 
* How you installed LSP (Package Control or from git?)
  * Package Control
* Minimal reproduction steps
  * create a minimal php package with autoloading
  * create a `src/First` and `src/Second` folders
  * add an App class with the appropriate namespaces, so we have e.g. `Acme\First\App` and `Acme\Second\App`
  * In a different class e.g. `Acme/Start` try writing App in a function. It will show two App instance, `Acme\First\App` and `Acme\Second\App`.
  * Select the second one
  * It will use the first App's additionalTextEdits and adds `use Acme\First\App` when I clearly selected the second (tried using tab or enter, doesnt matter)
* Log
```
{'textEdit': {'range': {'end': {'character': 4, 'line': 6}, 'start': {'character': 2, 'line': 6}}, 'newText': 'App'}, 'additionalTextEdits': [{'range': {'end': {'character': 0, 'line': 4}, 'start': {'character': 22, 'line': 2}}, 'newText': '\n\nuse Acme\\Test\\App;\n\n'}], 'kind': 7, 'detail': 'use Acme\\Test\\App', 'sortText': 'App', 'label': 'App', 'data': 1264852955825781},
{'textEdit': {'range': {'end': {'character': 4, 'line': 6}, 'start': {'character': 2, 'line': 6}}, 'newText': 'App'}, 'additionalTextEdits': [{'range': {'end': {'character': 0, 'line': 4}, 'start': {'character': 22, 'line': 2}}, 'newText': '\n\nuse Acme\\Third\\App;\n\n'}], 'kind': 7, 'detail': 'use Acme\\Third\\App', 'sortText': 'App', 'label': 'App', 'data': 4436429581058069}
```

From the logs, it's clearly seen that the use expression is correct for the different classes. It just that it always uses the first one and there is no way to import the second namespace

**Comment by tomv564** [comment]
Sorry, but this cannot be fixed with sublime’s completion API - it does not tell us which completion item was chosen/inserted. Our workaround does a lookup by inserted string, but obviously cannot differentiate between your two ‘App’ imports.

Thanks for the detailed report, hope it’s workable for you otherwise!

**Comment by predragnikolic** [comment]
Can the new Sublime api for autocompleton handle this? If so can we reopen the issue?

**Comment by ayoub-benali** [comment]
Yes the new API allows to fix this issue. I would re-open as well but label it for ST4

**Comment by tomv564** [comment]
Good point, I guess you are referring to using a command with the completion item?

I like to keep the issue list as relevant as possible, so I personally don't see the point of opening this issue until manpower shows up to work on ST4000 features.

**Comment by ayoub-benali** [comment]
yes the completion item. I am willing to work on this topic, can we have a `st4000` branch along side master ?

**Comment by tomv564** [comment]
Yes, feel free to create a PR  to the `st4000-exploration` branch.

**Comment by ayoub-benali** [comment]
Thanks

**Comment by rwols** [comment]
Supposedly fixed by #866 (won't be available for ST3).
